packages/testlab/src/test-sandbox.ts

```diff
@@ -37,19 +38,36 @@ export class TestSandbox {
    * Will create a directory if it doesn't already exist. If it exists, you
    * still get an instance of the TestSandbox.
    *
-   * @param path - Path of the TestSandbox. If relative (it will be resolved relative to cwd()).
-   */
-  constructor(path: string) {
-    // resolve ensures path is absolute / makes it absolute (relative to cwd())
-    this._path = resolve(path);
-    ensureDirSync(this.path);
-  }
-
-  /**
-   * Returns the path of the TestSandbox
+   * @example
+   * ```ts
+   * // Create a sandbox as a unique temporary subdirectory under the rootPath
+   * const sandbox = new TestSandbox(rootPath);
+   *
+   * // Create a sandbox in the root path directly
+   * // This is same as the old behavior
+   * const sandbox = new TestSandbox(rootPath, '');
+   * const sandbox = new TestSandbox(rootPath, '.');
+   *
+   * // Create a sandbox in the `test1` subdirectory of the root path
+   * const sandbox = new TestSandbox(rootPath, 'test1');
```

Comment:
I have mixed feeling about this API, I find it a bit difficult to understand the intent when reading code creating new sandboxes. I feel the reader needs a detailed knowledge of TestSandbox behavior to be able to understand what the second parameter means and how it changes sandbox behavior.

How about introducing a new `options` argument?

```ts
// Create a sandbox as a unique temporary subdirectory under the rootPath
const sandbox = new TestSandbox(rootPath);

const sandbox = new TestSandbox(rootPath, {
  unique: true,
});

// Create a sandbox in the root path directly
// This is same as the old behavior
const sandbox = new TestSandbox(rootPath, {
  unique: false,
});

const sandbox = new TestSandbox(rootPath, {
  subdir: '.', // implies unique: false
});

// Create a sandbox in the `test1` subdirectory of the root path
const sandbox = new TestSandbox(rootPath, {
  subdir: 'test1', // implies unique: false
});
```

Reply:
Improved with `TestSandboxOptions` as `{subdir: boolean | string}`.